operator/api/v1alpha1/keptnworkloadinstance_types.go

```diff
@@ -63,30 +79,15 @@ func init() {
 	SchemeBuilder.Register(&KeptnWorkloadInstance{}, &KeptnWorkloadInstanceList{})
 }
 
-type WorkloadInstancePhase string
-
-const (
-	// WorkloadInstancePending means that none of the WorkloadInstances have been created.
-	WorkloadInstancePending WorkloadInstancePhase = "Pending"
-	// WorkloadInstanceRunning means that all of the WorkloadInstances have been started.
-	WorkloadInstanceRunning WorkloadInstancePhase = "Running"
-	// WorkloadInstanceSucceeded means that all of the WorkloadInstances have been finished successfully.
-	WorkloadInstanceSucceeded WorkloadInstancePhase = "Succeeded"
-	// WorkloadInstanceFailed means that one or more pre-deployment checks was not successful and terminated.
-	WorkloadInstanceFailed WorkloadInstancePhase = "Failed"
-	// WorkloadInstanceUnknown means that for some reason the state of the application could not be obtained.
-	WorkloadInstanceUnknown WorkloadInstancePhase = "Unknown"
-)
-
-func (i KeptnWorkloadInstance) IsCompleted() bool {
-	if i.Status.PreDeploymentPhase == WorkloadInstanceSucceeded || i.Status.PreDeploymentPhase == WorkloadInstanceFailed || i.Status.PreDeploymentPhase == WorkloadInstanceUnknown {
+func (i KeptnWorkloadInstance) IsPreDeploymentCompleted() bool {
+	if i.Status.PreDeploymentStatus == common.StateSucceeded || i.Status.PreDeploymentStatus == common.StateFailed || i.Status.PreDeploymentStatus == common.StateUnknown {
```

Comment:
since we're using common.Status for pre/post etc. we could provide a `IsCompleted()` method for the common.Status struct